packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -51,21 +50,22 @@ function mergeInitialState(
  */
 function inputControlStateReducer(
 	composedStateReducers: StateReducer
-): LeadStateReducer {
+): StateReducer {
 	return ( state, action ) => {
-		if ( ! ( 'type' in action ) ) {
-			// Returns the new state without running other reducers. These are
-			// controlled updates from props and need no specialization.
-			return {
-				...state,
-				value: `${ action.value ?? '' }`,
-				isDirty: false,
-				_event: undefined,
-			};
-		}
 		const nextState = { ...state };
 
 		switch ( action.type ) {
+			/*
+			 * Controlled updates
+			 */
+			case actions.CONTROL:
+				nextState.value = `${ action.payload.value }`;
```

Comment:
I suppose that when used in a type-checked environment no one will ever provide anything but a string as `InputControl`’s `value` but if we'd like it to ensure it's actually a string at runtime this seemed good.

With your mention of the other actions, it makes me realize they could also be a place that non-string types are slipped in at runtime by an extending reducer and so if we want to ensure string type at runtime we could look at doing it in a single place before the base reducer returns so it would cover all actions.

I'm happy to apply your suggestion and save this for a follow up (or not at all if it doesn't seem worthy of concern).

Reply:
> With your mention of the other actions, it makes me realize they could also be a place that non-string types are slipped in at runtime by an extending reducer and so if we want to ensure string type at runtime we could look at doing it in a single place before the base reducer returns so it would cover all actions.

For the sake of consistency, I believe that we should apply the same strategy to all actions — so this option sounds like a good idea.

> I'm happy to apply your suggestion and save this for a follow up (or not at all if it doesn't seem worthy of concern).

I'd rather do this in a follow-up, so that we keep our PRs small and focused

Reply:
👍 1456a56